src/Tizen.System.PowerUsage/PowerUsage/PowerUsage.cs

```diff
@@ -0,0 +1,242 @@
+        /// <exception cref="InvalidOperationException">In case of any system error.</exception>
+        /// <exception cref="NotSupportedException">In case power usage is not supported</exception>
+        public static double GetPowerUsage(string appID, DateTime start, DateTime end)
+        {
+            double batteryUsage = 0;
+            DateTimeOffset startTime = DateTime.SpecifyKind(start, DateTimeKind.Utc);
+            DateTimeOffset endTime = DateTime.SpecifyKind(end, DateTimeKind.Utc);
+            PowerUsageError ret = (PowerUsageError)Interop.PowerUsage.GetPowerUsageByApp(appID, startTime.ToUnixTimeSeconds(), endTime.ToUnixTimeSeconds(), out batteryUsage);
+            if (ret != PowerUsageError.None)
+            {
+                Log.Error(PowerUsageErrorFactory.LogTag, "Error getting battery usage by app," + ret);
+                throw PowerUsageErrorFactory.ThrowPowerUsageException(ret);
+            }
+            return batteryUsage;
+        }
+
+        /// <summary>
+        /// Gets the battery usage in mAh(milli-Ampere hour) by a resource for certain time interval.
+        /// </summary>
+        /// <since_tizen> 7 </since_tizen>
+        /// <param name="rtype">Identifier of resource type. BLE, WiFi, CPU etc.</param>
+        /// <param name="start">Start Time for data in DateTime.</param>
+        /// <param name="end">End Time for data in DateTime.</param>
+        /// <returns>Returns the battery usage in mAh(milli-Ampere hour) by a resource for certain time interval.</returns>
+        /// <feature>http://tizen.org/feature/battery</feature>
+        /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+        /// <exception cref="ArgumentException">When an invalid parameter value is set.</exception>
+        /// <exception cref="UnauthorizedAccessException">If the privilege is not set.</exception>
+        /// <exception cref="InvalidOperationException">In case of any system error.</exception>
+        /// <exception cref="NotSupportedException">In case power usage is not supported</exception>
+        public static double GetPowerUsage(PowerUsageResourceType rtype, DateTime start, DateTime end)
+        {
+            double batteryUsage = 0;
+            DateTimeOffset startTime = DateTime.SpecifyKind(start, DateTimeKind.Utc);
+            DateTimeOffset endTime = DateTime.SpecifyKind(end, DateTimeKind.Utc);
+
+            PowerUsageError ret = (PowerUsageError)Interop.PowerUsage.GetPowerUsageByResource(rtype, startTime.ToUnixTimeSeconds(), endTime.ToUnixTimeSeconds(), out batteryUsage);
+            if (ret != PowerUsageError.None)
+            {
+                if (ret == PowerUsageError.RecordNotFound)
+                {
+                    Log.Error(PowerUsageErrorFactory.LogTag, "Error PowerUsageResourceType is not supported");
+                    throw new ArgumentException($"{rtype} is not supproted", nameof(rtype));
```

Comment:
how about `NotSupportedException` instead of `ArgumentException`

Reply:
NotSupportedException  is used to check whether power usage is supported  by the device or not.

In this case, if PowerUsageResourceType is not supported, then InvalidOperationException is thrown, but while using this api, user knows the resource type he is sending, so incase  of unsupported resource, we are taking as wrong arguments and throwing Argument Exception